sherpa/plot/__init__.py

```diff
@@ -564,13 +585,35 @@ def plottop(self, plot, *args, **kwargs):
         clearaxes = kwargs.get('clearwindow', True)
         self._clear_window(0, 0, clearaxes)
 
+        """
         # FIXME: should not know about FitPlot, terrible hack to remove label
         if isinstance(plot, FitPlot):
             plot.dataplot.xlabel = ''
             plot.modelplot.xlabel = ''
         else:
             plot.xlabel = ''
 
+        """
+
+        # Slightly-more Pythonic way to clear the labels than checking
+        # the class of the object.
+        #
+        # The order here could matter if things get changed (such as
+        # FitPlot being re-worked to use a common labelling system)
+        #
+        # Unlike the previous version, this will not fail (here at least)
```

Comment:
It is good that you check for the attribute before accessing it, but the here comment is superfluous.

Reply:
Okay - I'll remove it then.